webapp/test_results_handler.go

```diff
@@ -90,36 +82,46 @@ func parseTestResultsUIFilter(r *http.Request) (filter testResultsUIFilter, err
 	}
 	ctx := shared.NewAppEngineContext(r)
 
-	if testRunFilter.IsDefaultQuery() {
-		experimentalByDefault := shared.IsFeatureEnabled(ctx, "experimentalByDefault")
-		experimentalAlignedExceptEdge := shared.IsFeatureEnabled(ctx, "experimentalAlignedExceptEdge")
-		masterRunsOnly := shared.IsFeatureEnabled(ctx, "masterRunsOnly")
-		if experimentalByDefault {
-			if experimentalAlignedExceptEdge {
-				testRunFilter = testRunFilter.OrAlignedExperimentalRunsExceptEdge()
+	filter.PR, err = shared.ParsePRParam(r)
+	if err != nil {
+		return filter, err
+	} else if filter.PR != nil {
+		one := 1
+		filter.MaxCount = &one
+	} else {
+		if testRunFilter.IsDefaultQuery() {
+			experimentalByDefault := shared.IsFeatureEnabled(ctx, "experimentalByDefault")
+			experimentalAlignedExceptEdge := shared.IsFeatureEnabled(ctx, "experimentalAlignedExceptEdge")
+			masterRunsOnly := shared.IsFeatureEnabled(ctx, "masterRunsOnly")
+			if experimentalByDefault {
+				if experimentalAlignedExceptEdge {
+					testRunFilter = testRunFilter.OrAlignedExperimentalRunsExceptEdge()
+				} else {
+					testRunFilter = testRunFilter.OrExperimentalRuns()
+				}
 			} else {
-				testRunFilter = testRunFilter.OrExperimentalRuns()
+				testRunFilter = testRunFilter.OrAlignedStableRuns()
+			}
+			if masterRunsOnly {
+				testRunFilter = testRunFilter.MasterOnly()
 			}
-		} else {
-			testRunFilter = testRunFilter.OrAlignedStableRuns()
-		}
-		if masterRunsOnly {
-			testRunFilter = testRunFilter.MasterOnly()
 		}
-	}
 
-	filter.testRunUIFilter = parseTestRunUIFilter(testRunFilter)
+		filter.testRunUIFilter = convertTestRunUIFilter(testRunFilter)
+	}
 
 	diff, err := shared.ParseBooleanParam(r, "diff")
 	if err != nil {
 		return filter, err
 	}
 	filter.Diff = diff != nil && *diff
-	diffFilter, _, err := shared.ParseDiffFilterParams(r)
-	if err != nil {
-		return filter, err
+	if filter.Diff {
```

Comment:
I can't quite tell if anything in this PR made this change necessary, but looks reasonable :)

Reply:
Drive-by cleanup.